app/pages/Accounts/VerifyAddress.tsx

```diff
@@ -1,65 +1,81 @@
-import Card from '~/cross-app-components/Card';
 import MessageModal from '~/components/MessageModal';
 import Ledger from '~/components/ledger/Ledger';
 import { asyncNoOp } from '~/utils/basicHelpers';
-import findLocalDeployedCredentialWithWallet from '~/utils/credentialHelper';
+import {
+    findLocalDeployedCredential,
+    hasFirstCredential,
+} from '~/utils/credentialHelper';
 import { specificIdentitySelector } from '~/features/IdentitySlice';
+import { walletIdSelector } from '~/features/WalletSlice';
 import { Account, ClassName } from '~/utils/types';
-
-import styles from './Accounts.module.scss';
+import { useAsyncMemo } from '~/utils/hooks';
 
 interface Props extends ClassName {
     account: Account;
+    display: ReactElement;
 }
 
 /**
  * Allows the user to verify the address of the current account, on the ledger.
  */
-export default function VerifyAddress({ account, className }: Props) {
+export default function VerifyAddress({ account, display, className }: Props) {
     const [opened, setOpened] = useState(false);
     const [warning, setWarning] = useState<string>();
     const identity = useSelector(specificIdentitySelector(account.identityId));
+    const walletId = useSelector(walletIdSelector);
+    const credential = useAsyncMemo(
+        () =>
+            walletId
+                ? findLocalDeployedCredential(walletId, account.address)
+                : Promise.resolve(undefined),
+        () => {},
+        [walletId, account.address]
+    );
 
-    async function ledgerCall(ledger: ConcordiumLedgerClient) {
-        if (identity === undefined) {
-            throw new Error(
-                'No identity was found for this account. This is an internal error that should be reported.'
-            );
-        }
-
-        const credential = await findLocalDeployedCredentialWithWallet(
-            account.address,
-            ledger
-        );
-
-        if (credential === undefined) {
-            throw new Error(
-                'No credentials, that were created by the current Ledger, were found on this account. Please verify that the connected Ledger is for this account.'
-            );
-        }
+    const reasonForNotBeingVerifiable = useAsyncMemo(
+        async () => {
+            if (identity === undefined) {
+                return 'No identity was found for this account. This is an internal error that should be reported.';
+            }
+            if (!(await hasFirstCredential(account.address))) {
+                return "An account's address is derived from the initial credential on that account. The initial credential is not in this wallet, and therefore it is not possible to verify the address of the account.";
+            }
+            if (identity.version === 0) {
+                return 'The identity used to create this account uses a deprecated version of the key generation. It is recommended to create a new identity and use accounts on that instead.';
```

Comment:
I am not certain that we should recommend people to do this.

Reply:
Changed it to:
```
The address cannot be verified, 
because the identity used to create this account uses a deprecated version of the key generation, 
which is not available on the ledger.
```